test/test_autograd.py

```diff
@@ -4946,7 +4946,113 @@ def test_autograd_simple_views_python(self):
+            base = torch.rand(2, requires_grad=requires_grad).clone()
+            # 1. Create a view with `grad_mode=grad_mode_view`
+            with torch.set_grad_enabled(grad_mode_view):
+                if fn_type == "multi_view":
+                    inp = base.unbind()[0]
+                elif fn_type == "custom" :
+                    inp = inplace_custom(base)
+                else:
+                    inp = base.view_as(base)
+
+            # 2. Perform inplace view with `grad_mode=grad_mode_iview`
+            with torch.set_grad_enabled(grad_mode_iview):
+                if error1 is not None:
+                    with self.assertRaisesRegex(RuntimeError, error1):
+                        fn(inp)
+                    return
+                else:
+                    fn(inp)
+            # 3. Do inplace on the (new) view
+            if error2 is not None:
+                with self.assertRaisesRegex(RuntimeError, error2):
+                    inp.add_(1)
+            else:
+                inp.add_(1)
+
+        no_grad_err = "A view was created in no_grad mode"
+        multi_view_err = "function that returns multiple views"
+        custom_err = "view was created inside a custom Function"
+
+        def run_tests(fn):
+            for fn_type in ("normal", "multi_view", "custom"):
+                for grad_mode_view in (True, False):
+                    for grad_mode_iview in (True, False):
+                        for requires_grad in (True, False):
+                            error1 = None  # expected error when we do inplace_view on original view
+                            error2 = None  # expected error when we do inplace on the resulting view
+
+                            if not grad_mode_view and grad_mode_iview and requires_grad:
+                                error1 = no_grad_err
+                            if not grad_mode_view and not grad_mode_iview and requires_grad:
+                                error2 = no_grad_err
+
+                            if fn_type == "multi_view":
+                                if grad_mode_view and grad_mode_iview and requires_grad:
+                                    error1 = multi_view_err
+                                if grad_mode_view and not grad_mode_iview and requires_grad:
+                                    error2 = multi_view_err
+
+                            if fn_type == "custom":
+                                if grad_mode_view and grad_mode_iview and requires_grad:
+                                    error1 = custom_err
+                                if grad_mode_view and not grad_mode_iview and requires_grad:
+                                    error2 = custom_err
+
+                            print(fn_type, grad_mode_view, grad_mode_iview, requires_grad, error1, error2)
+                            run_test(fn, fn_type, grad_mode_view, grad_mode_iview, requires_grad, error1, error2)
+
+        # TODO maybe we should just run all of theses?
```

Comment:
How long does this take?
If it is less than a couple seconds, you should definitely run all of them.

Reply:
It is pretty fast. Around 0.2s